packages/funding-rate-proposer/index.js

```diff
@@ -0,0 +1,180 @@
+    // properties that are shared across all of these new pricefeeds.
+    const fundingRateProposer = new FundingRateProposer({
+      logger,
+      perpetualFactoryClient,
+      gasEstimator,
+      account: accounts[0],
+      commonPriceFeedConfig,
+      perpetualProposerConfig
+    });
+
+    // Create a execution loop that will run indefinitely (or yield early if in serverless mode)
+    for (;;) {
+      await retry(
+        async () => {
+          await fundingRateProposer.update();
+          await fundingRateProposer.updateFundingRates();
+          return;
+        },
+        {
+          retries: errorRetries,
+          minTimeout: errorRetriesTimeout * 1000, // delay between retries in ms
+          randomize: false,
+          onRetry: error => {
+            logger.debug({
+              at: "PerpetualFundingRateProposer#index",
+              message: "An error was thrown in the execution loop - retrying",
+              error: typeof error === "string" ? new Error(error) : error
+            });
+          }
+        }
+      );
+      // If the polling delay is set to 0 then the script will terminate the bot after one full run.
+      if (pollingDelay === 0) {
+        logger.debug({
+          at: "PerpetualFundingRateProposer#index",
+          message: "End of serverless execution loop - terminating process"
+        });
+        await waitForLogger(logger);
+        break;
+      }
+      logger.debug({
+        at: "PerpetualFundingRateProposer#index",
+        message: "End of execution loop - waiting polling delay",
+        pollingDelay: `${pollingDelay} (s)`
+      });
+      await delay(Number(pollingDelay));
+    }
+  } catch (error) {
+    // If any error is thrown, catch it and bubble up to the main try-catch for error processing in the Poll function.
+    throw typeof error === "string" ? new Error(error) : error;
+  }
+}
+
+async function Poll(callback) {
+  try {
+    // This object is spread when calling the `run` function below. It relies on the object enumeration order and must
+    // match the order of parameters defined in the`run` function.
+    const executionParameters = {
+      // Default to 1 minute delay. If set to 0 in env variables then the script will exit after full execution.
+      pollingDelay: process.env.POLLING_DELAY ? Number(process.env.POLLING_DELAY) : 60,
+      // Default to 3 re-tries on error within the execution loop.
+      errorRetries: process.env.ERROR_RETRIES ? Number(process.env.ERROR_RETRIES) : 5,
+      // Default to 10 seconds in between error re-tries.
+      errorRetriesTimeout: process.env.ERROR_RETRIES_TIMEOUT ? Number(process.env.ERROR_RETRIES_TIMEOUT) : 10,
+      // Common price feed configuration passed along to all those constructed by proposer.
+      commonPriceFeedConfig: process.env.COMMON_PRICE_FEED_CONFIG
+        ? JSON.parse(process.env.COMMON_PRICE_FEED_CONFIG)
+        : { lookback: 7200 },
```

Comment:
Hm should we? Sholdn't it be inferred from the OptimisticOracle's proposal liveness? Regardless I'll add a TODO

Reply:
Actually, I'm not sure there's any sort of lookback that makes sense, because the funding rate proposer is always checking out the current price of the funding rate, never historical. WDYT

Reply:
Yep, you're right. Nice catch. We probably don't want to set it to 0 or anything, so just leaving the default seems right.

Reply:
Bump.

Reply:
I removed this